src/ResourceManager/ManagedServiceIdentity/Commands.ManagedServiceIdentity/UserAssignedIdentities/GetAzureRmUserAssignedIdentityCmdlet.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ManagedServiceIdentity.Common;
+using Microsoft.Azure.Commands.ManagedServiceIdentity.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.ManagedServiceIdentity.UserAssignedIdentities
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmUserAssignedIdentity", DefaultParameterSetName = SubscriptionParameterSet)]
+    [OutputType(typeof (PsUserAssignedIdentity))]
+    public class GetAzureRmUserAssignedIdentityCmdlet : MsiBaseCmdlet
+    {
+        private const string SubscriptionParameterSet = "SubscriptionParameterSet";
```

Comment:
You don't need three parameter sets to represent this.  Have one parameter set with no parameters (as default), and one parameter set with ResourceGroupName mandatory and Name optional.  It might also be cool to allow the user to just specific name and not resourcegroupname, and you would run listbysubscription and filter based on the name.  That is your call though.